### ISSUE TITLE
[Bug] Attachments to Service Records

### ISSUE THREAD

**Issue by rtrickett337** [hargata/lubelog#128]
When adding a .PDF (specifically an oil sample report) either as a new service record or editing existing, when clicking the add or edit button after uploading, there's no response. The dialogue persists and I have to remove the attachment to edit or add the service record. When I go into the mapped volume for the docker image, I see the .PDF saved in the file directory, but there's no reference to it within the service record in Lubelogger.

**Comment by hargata** [comment]
was it a pretty big pdf file?

**Comment by rtrickett337** [comment]
no...20kb

**Comment by hargata** [comment]
I see, let me look into this. I'll get back to you by end of today if I can re-produce the bug.

**Comment by rtrickett337** [comment]
No worries and thanks for the quick response! Love the app, by the way! I too had an excel spreadsheet for each vehicle and really like the idea of modernizing and being able to self host this!

**Comment by rtrickett337** [comment]
For reference, I'm running this on a Synology NAS in docker. I use Portainer to manage my docker containers and setup Lubelogger using the Docker Setup (GHCR) method / docker-compose. In Portainer I deployed the stack by pulling from the git repository. Once I deployed the stack, I then updated the volumes to ensure these were persistent and re-deployed the container. Here's what my mapping looks like:
<img width="707" alt="Screenshot 2024-01-20 at 9 00 12 AM" src="https://github.com/hargata/lubelog/assets/73918799/419ebae9-7da0-4f3c-8140-24e28aeebc0e">

**Comment by rtrickett337** [comment]
I also noticed the same behavior is occurring when I try and add a vehicle image. I'm trying to add .PNG file that's ~550kb. It's uploading/saving to the docker volume, but not showing in Lubelogger.

**Comment by hargata** [comment]
do you see any errors when you pull up Chrome console by pressing F12?

**Comment by hargata** [comment]
I also noticed that your volumes seem to all be mapped to the same directory, can you try mapping them to a subfolder within the directory?

**Comment by rtrickett337** [comment]
Here's what I get when I try and add:

<img width="556" alt="Screenshot 2024-01-20 at 9 22 06 AM" src="https://github.com/hargata/lubelog/assets/73918799/f80c8680-ea40-4be9-9580-2624ab34f86d">

**Comment by hargata** [comment]
I think I know what's going on. When you upload a file, it first gets uploaded to the /temp folder, and then when the save/edit was confirmed, it moves the file out of the temp folder into /images or /documents. 
Since temp is mapped into the same volume as images and documents, it's trying to move the file into the same directory it's in, which causes it to break.

Try mapping just /temp into its own volume and it should start working.

**Comment by rtrickett337** [comment]
Ah! Thanks. I'll try that. I'm thinking /temp doesn't need to be persistent, and could just be left as-is when the container is compiled and that volume is within the container.

I will clean up the persistent folder though so that it has some structure.

**Comment by rtrickett337** [comment]
Ok, that fixed it! Thank you. Below I shared how I reconfigured the container and the folders I created in my docker/app folder. Perhaps I could've done something different with the way I ran the docker-compose to create the persistent volumes and folders right off the bat. I'm not used to having to specify a volume map for each folder need. My experience has been I just have to specify the one map/bind to ensure it's persistent and when I deploy the container, any subfolders needed are automatically created by the app. No worries though, learned something today.

<img width="839" alt="Screenshot 2024-01-20 at 9 46 53 AM" src="https://github.com/hargata/lubelog/assets/73918799/a2f82102-6b83-49ef-b571-dfe0ddaf1ded">

I also see that you've added some new features this morning as well :)
